Лаб. 3 6412 Ноздрякова Марина

kaaatz-e opened the pull request:
**ФИО:** Ноздрякова Марина
**Номер группы:** 6412
**Номер лабораторной:** 2,3
**Номер варианта:** 40
**Краткое описание предметной области:** Авиакомпания
**Краткое описание добавленных фич:** Реализовано серверное приложение, осуществляющее базовые CRUD-операции. Реализован доступ к базе данных MongoDb через Entity Framework Core. Запуск приложения и управление инфраструктурой организованы с помощью Aspire.

kolyanch reviewed (changes requested):
В клиенте отсутствует валидация существования связанных сущностей. При вводе предприятия, если введен несуществующий тип предприятия, приложение падает с ошибкой. Необходимо выдать пользователю user-friendly сообщение

kaaatz-e committed:
add entity classes for airline domain model
feat: create AircraftFamily, AircraftModel, Flight, Passenger, Ticket

kaaatz-e committed:
docs: add XML documentation to domain entities

kaaatz-e committed:
feat: add DataSeeder
refactor: move domain entities to separate AirCompany.Domain/Entities folder

kaaatz-e committed:
test: add unit tests

kaaatz-e committed:
docs: add add XML documentation to DataSeeder

kaaatz-e committed:
docs: add XML documentation to unit tests

kaaatz-e committed:
docs: remove unnecessary links

kaaatz-e committed:
ci: add GitHub Actions workflow for .NET tests

kaaatz-e committed:
docs: add ER diagram for air company domain model

kaaatz-e committed:
refactor: migrate ID types from int to Guid
refactor: update DataSeeder to use auto-generated Guid IDs

kaaatz-e committed:
refactor: replace fields with auto-properties
refactor: initialize properties directly in constructor

kaaatz-e committed:
test: refactor order validation in unit tests

kaaatz-e committed:
test: fix Assert usage from collection to single object

kaaatz-e committed:
test: delete unnecessary test

kaaatz-e committed:
feat: add MongoDB Entity Framework Core infrastructure and repositories

kaaatz-e committed:
feat: add dtos
feat: add services and mapper profile
feat: add controllers and Program.cs
feat: add AppHost.cs

kaaatz-e committed:
feat: add FK in entities
fix: DataSeeder
fix: tests
refactor: recreate projects
refactor: replace navigation property with foreign key

kaaatz-e committed:
docs: add add XML documentation

alxmcs commented on AirCompany/AirCompany.Api.Host/AirCompany.Api.Host.http:
Это файл с дефолтным маршрутом к контроллеру, который генерится в шаблоне проекта по дефолту. Он не нужен

alxmcs reviewed (approved):
Мне все понравилось с первого раза :+1: :+1: :+1:
Только лишний файл удали из репозитория

kaaatz-e committed:
chore: delete .http file
